realm/src/main/java/io/realm/Realm.java

```diff
@@ -657,6 +658,14 @@ private static void validateAgainstExistingConfigurations(RealmConfiguration new
                 throw new IllegalArgumentException("Two configurations with different schemas are trying to open " +
                         "the same Realm file. Their schema must be the same: " + newConfiguration.getPath());
             }
+
+            // Check if the durability is the same
+            SharedGroup.Durability cachedDurability = cachedConfiguration.getDurability();
+            SharedGroup.Durability newDurability = newConfiguration.getDurability();
+            if (!cachedDurability.equals(newDurability)) {
+                throw new IllegalArgumentException("A Realm cannot be both in-memory and persisted. Two conflicting " +
+                        "configurations pointing to " + newConfiguration.getRealmFileName() + " are being used.");
```

Comment:
That remind me that do we need to restrict use from using DEFAULT_REALM_NAME to create in-memory Realm? It won't be harmful at all, but it has slightly differnt API with IOS which always ask user to give the in-memory Realm a name (Although it can be default if user wants.)